### PR TITLE
Postgres Plugin with documentation

### PR DESCRIPTION
**Pull request by aussendorf**
Has also modified BaseClass and LocalFileset Plugin as in PR485

### CODE REVIEW

File: core/src/plugins/filed/BareosFdPluginPostgres.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        except:
+            bareosfd.JobMessage(
+                context,
+                bJobMessageType["M_FATAL"],
+                "Could not connect to database %s, user %s\n"
+                % (self.dbname, self.dbuser),
+            )
+            return bRCs["bRC_Error"]
+        # If level is not Full, we only backup newer WAL files
+        if chr(self.level) == "F":
```
Comment (franku) [inline core/src/plugins/filed/BareosFdPluginPostgres.py:154]: This does not match the comment.

File: core/src/plugins/filed/BareosFdPluginPostgres.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+#
+# Author: Maik Aussendorf
+#
+# Bareos python plugins class that adds files from a local list to
+# the backup fileset
```
Comment (joergsteffens) [inline core/src/plugins/filed/BareosFdPluginPostgres.py:25]: This comment does not describe the functionality.

File: core/src/plugins/filed/BareosFdPluginPostgres.py [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+from BareosFdPluginBaseclass import *
+
+
+class BareosFdPluginPostgres(
+    BareosFdPluginLocalFileset.BareosFdPluginLocalFileset
+):  # noqa
+    """
+    Simple Bareos-FD-Plugin-Class that parses a file and backups all files
+    listed there Filename is taken from plugin argument 'filename'
+    """
```
Comment (joergsteffens) [inline core/src/plugins/filed/BareosFdPluginPostgres.py:48]: This comment does not describe the functionality.

File: docs/manuals/source/TasksAndConcepts/Plugins.rst [hunk 4]
```diff
@@ -1164,7 +1165,156 @@ If things don't work as expected, make sure that
 - Make sure *XtraBackup* works as user root, MySQL access needs to be
   configured properly
 
+PostgreSQL Plugin
+~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (joergsteffens) [inline docs/manuals/source/TasksAndConcepts/Plugins.rst:1169]:
```suggestion
~~~~~~~~~~~~~~~~~
```
Sphinx will complain about to long decoration.

File: docs/manuals/source/TasksAndConcepts/Plugins.rst [hunk 5]
```diff
@@ -1164,7 +1165,156 @@ If things don't work as expected, make sure that
...
+* Backups that complete quickly and reliably
+* Uninterrupted transaction processing during backups
+* Savings on disk space and network bandwidth
+* Higher uptime due to faster restore time
+
+Requires |postgresql| Version 9.x or newer.
+
+
+Concept
+^^^^^^^^^^^^^
```
Comment (joergsteffens) [inline docs/manuals/source/TasksAndConcepts/Plugins.rst:1189]:
```suggestion
^^^^^^^
```

File: core/src/plugins/filed/BareosFdPluginPostgres.py [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+                switchLsnStmt = "SELECT pg_switch_wal()"
+            elif pgMajorVersion >= 9:
+                getLsnStmt = "SELECT pg_current_xlog_location()"
+                switchLsnStmt = "SELECT pg_switch_xlog()"
+            if pgMajorVersion < 9:
+                bareosfd.JobMessage(
+                    context,
+                    bJobMessageType["M_INFO"],
+                    "WAL switching not supported on Postgres Version < 9\n",
+                )
```
Comment (joergsteffens) [inline core/src/plugins/filed/BareosFdPluginPostgres.py:181]: No return value here? Shouldn't it return FAILED or something?

File: systemtests/tests/python-fd-plugin-postgres-test/database/setup_local_db.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+# for online backups we need wal_archiving
+echo "wal_level = archive" >> data/postgresql.conf
+echo "archive_mode = on" >> data/postgresql.conf
+echo "archive_command = 'cp %p @current_test_directory@/database/wal_archive'" >> data/postgresql.conf
+echo "max_wal_senders = 10" >> data/postgresql.conf
+
+pg_ctl -D data -l  log/logfile start
+sleep 10 
+
+echo "CREATE ROLE root WITH SUPERUSER CREATEDB CREATEROLE REPLICATION LOGIN" |psql -h @current_test_directory@/database/tmp postgres
```
Comment (franku) [inline systemtests/tests/python-fd-plugin-postgres-test/database/setup_local_db.sh:21]:
```suggestion
createuser root -s -h @current_test_directory@/database/tmp postgres

```

File: systemtests/tests/python-fd-plugin-postgres-test/python-modules/BareosFdPluginLocalFilesetWithRestoreObjects.py [hunk 8]
```diff
@@ -0,0 +1,353 @@
+#!/usr/bin/env python
```
Comment (franku) [inline systemtests/tests/python-fd-plugin-postgres-test/python-modules/BareosFdPluginLocalFilesetWithRestoreObjects.py:1]: Is this file needed anymore? This may be a copy-relict of a sample.

File: systemtests/tests/python-fd-plugin-postgres-test/python-modules/bareos-fd-local-fileset-with-restoreobjects.py [hunk 9]
```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
```
Comment (franku) [inline systemtests/tests/python-fd-plugin-postgres-test/python-modules/bareos-fd-local-fileset-with-restoreobjects.py:1]: Also consider to delete this file.

File: core/src/plugins/filed/BareosFdPluginPostgres.py (hunk 6 above)
Comment (aussendorf) [reply core/src/plugins/filed/BareosFdPluginPostgres.py:202]: Backup itself does still work. It's only the forced wal-switch before an incremental backup is not supported before version 9. It is just this option not working.

File: core/src/plugins/filed/BareosFdPluginPostgres.py (hunk 2 above)
Comment (aussendorf) [reply core/src/plugins/filed/BareosFdPluginPostgres.py]: fixed with latest commit

File: core/src/plugins/filed/BareosFdPluginPostgres.py (hunk 3 above)
Comment (aussendorf) [reply core/src/plugins/filed/BareosFdPluginPostgres.py:48]: fixed with latest commit

File: systemtests/tests/python-fd-plugin-postgres-test/python-modules/BareosFdPluginLocalFilesetWithRestoreObjects.py (hunk 8 above)
Comment (aussendorf) [reply systemtests/tests/python-fd-plugin-postgres-test/python-modules/BareosFdPluginLocalFilesetWithRestoreObjects.py:1]: You are probably right but this file is not part of the PR?

File: systemtests/tests/python-fd-plugin-postgres-test/python-modules/bareos-fd-local-fileset-with-restoreobjects.py (hunk 9 above)
Comment (aussendorf) [reply systemtests/tests/python-fd-plugin-postgres-test/python-modules/bareos-fd-local-fileset-with-restoreobjects.py:1]: You are probably right but this file is not part of the PR?

File: systemtests/tests/python-fd-plugin-postgres-test/python-modules/BareosFdPluginLocalFilesetWithRestoreObjects.py (hunk 8 above)
Comment (franku) [reply systemtests/tests/python-fd-plugin-postgres-test/python-modules/BareosFdPluginLocalFilesetWithRestoreObjects.py:1]: This file is part of this PR, because it appears in the file list.

File: systemtests/tests/python-fd-plugin-postgres-test/python-modules/bareos-fd-local-fileset-with-restoreobjects.py (hunk 9 above)
Comment (franku) [reply systemtests/tests/python-fd-plugin-postgres-test/python-modules/bareos-fd-local-fileset-with-restoreobjects.py:1]: This file is part of this PR, because it appears in the file list.

**Comment by franku** [comment]
This one is outdated.
